CMakeLists.txt

```diff
@@ -233,6 +233,7 @@ else()
   endif()
   add_compile_flag("-Wextra")
   add_compile_flag("-Wno-unused-parameter")
+  add_compile_flag("-Wno-unused-variable")
```

Comment:
Wait.. it seems that some of this pattern has already entered the codebase and it didn't cause any issues with any of the compilers that we use in CI.  For example here `func` is unused:

https://github.com/WebAssembly/binaryen/blob/3c1c8ae829d0026024de55aa65dc3d49bd8fd37a/src/passes/GlobalRefining.cpp#L57-L61

But neither clang nor gcc nor msvc seem to care.

I didn't just run into an issues where an old version of gcc-7 does complain though:

https://app.circleci.com/pipelines/github/emscripten-core/emsdk/1385/workflows/d696706d-f612-429d-94ae-12b0742e7cfb/jobs/7067?invite=true#step-105-303

So I guess my questions is, under what compiler are you seeing a warning for this?

Reply:
```
$ cat test.cc
#include <map>
#include <cstdio>

std::map<int, int>  map;

void foo() {
  for (auto& [key, val] : map) {
    printf("%d\n", key);
  }
}
$ clang++ -std=c++17 test.cc -c -Wall -Wextra -Werror
$ g++ -std=c++17 test.cc -c -Wall -Wextra -Werror
(no warnings shown)
```

Reply:
> So I guess my questions is, under what compiler are you seeing a warning for this?

It was on CI. But that was quite a long time ago. Maybe something has changed.

Reply:
> (no warnings shown)

Could you check the same for `auto& [key, _]`?

Reply:
Yup, still no warnings from modern clang/gcc.    g++-7 does show a warning but it seems we don't test with that in CI anywhere otherwise we would have seen failures for the existing instances

Reply:
So it is possible to remove this line from CMakeFile? As long as the minimum requirements are g++ >= 8 and clang >= 12 ?

Reply:
I'm not sure what about the clang version, or if it ever had this issue.    But in general yes I think you are correct.

Can you try this PR with this line removed?

Reply:
It looks like all builds passed successfully!